src/shmif/tui/tui_listwnd.c

```diff
@@ -227,8 +227,13 @@ static void key_input(struct tui_context* T, uint32_t keysym,
 	else if (keysym == TUIK_LEFT || keysym == TUIK_ESCAPE){
 			M->entry_state = -1;
 	}
-	else
-		;
+
+}
+
+static void input_mouse_button(struct tui_context* T, int last_x, int last_y, int button, bool active, int modifiers, void* tag) {
+	struct listwnd_meta* M = tag;
+	M->list_pos = last_y;
```

Comment:
NAK. This is (slightly) harder than it looks.

Nits: style: 80 columns, so br after * T, and BR before { for function prototype

Solution itself:
Mouse-button should actually set the entry_state/entry_pos and act as a proper select. The mouse_motion event handler would be better for modifying the list_pos modification and redraw, but even then three edge cases need to be handled:

1. clamp against list_sz (oob- deref on keyboard select otherwise)
2. list_pos is relative to the provided list, not the screen itself. If the list is on a different page or there’s hidden items in the list, you’d not get what you click on. See redraw() for how the screen-offset is calculated.
3. items marked as separators or inactive shouldn’t be selectable.

Notes:
the ‘double-trigger’ part is simply to check !active, you get two invocations, one for press, one for release. Typically want to mask release and trigger on press.